addon/components/au-file-upload.js

```diff
@@ -67,43 +69,53 @@ export default class AuFileUploadComponent extends Component {
       const response = yield file.upload(this.endPoint, {
         'Content-Type': 'multipart/form-data',
       });
-      const fileId = response.body?.data?.id;
+      const body = yield response.json();
+      const fileId = body?.data?.id;
       return fileId;
     } catch (e) {
-      this.uploadErrorData = [...this.uploadErrorData, { filename: file.name }];
+      this.addError(file);
       this.removeFileFromQueue(file);
       return null;
     }
   }
 
-  hasValidationErrors(file) {
-    // TODO: see if we can split the side-effects from the
-    // non-side-effects.  The name doesn't suggest this will yield a
-    // state change.
-    if (file.size > this.maxFileSizeMB * Math.pow(1024, 2)) {
-      this.uploadErrorData = [
-        ...this.uploadErrorData,
-        {
-          filename: file.name,
-          error: `Bestand is te groot (max ${this.maxFileSizeMB} MB)`,
-        },
-      ];
-      this.removeFileFromQueue(file);
-      return true;
+  @action
+  filter(file, files, index) {
+    let isFirstFile = index === 0;
+
+    if (isFirstFile) {
+      this.resetErrors();
+    } else {
+      if (!this.args.multiple) {
+        // We only upload the first file if `@multiple` is not set to true. This matches the behavior of ember-file-upload v4.
+        return false;
+      }
```

Comment:
I mimicked ember-file-upload v4's behavior where only the first file is uploaded if the users drops more than 1 and multiple is set to false. I'm not sure if this is the best behavior though. It might be better to not upload anything at all in that case?